kubernetes/nodecollection.go

```diff
@@ -145,10 +132,111 @@ func downloadNodeData(prefix string,
 	return failedNodeList, nil
 }
 
+// nodeFetchData is a convenience wrapper for
+// information used to fetch node stats and store
+// in the appropriate file location
+type nodeFetchData struct {
+	nodeName          string
+	prefix            string
+	workDir           *os.File
+	ClusterHostURL    string
+	containersRequest []byte
+}
+
+// directNodeFetch retrieves node stats directly from the node api
+func directNodeFetch(nodeSource NodeSource, config KubeAgentConfig, n *v1.Node, nd nodeFetchData) error {
+	ip, port, err := nodeSource.NodeAddress(n)
+	if err != nil {
+		return fmt.Errorf("problem getting node address: %s", err)
```

Comment:
I noticed that previously this error would have been `return nil, fmt.Errorf` in this case we would just be adding it as a failed node in the failedNodeList. I'm not sure what the intent was originally but i believe this would cause a different outcome. Feel free to correct me if I'm wrong!

Reply:
No, you're correct. I'm hoping someone who knew the intent of that error could weigh in, because it doesn't seem necessary to fail the whole node fetch if that part of the direct node fetch fails. In other cases we just move on to trying proxy, so I'm not sure why that error was being handled differently.